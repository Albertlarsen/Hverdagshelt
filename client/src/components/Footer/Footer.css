.footerClass{
    margin-top: 80px;
    height: 100px;
    width: 100%;
    text-align: center;
    background-color: #A0A0A3;

}