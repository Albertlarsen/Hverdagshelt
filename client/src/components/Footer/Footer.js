//@flow
import React, { Component } from 'react';
import {
  Grid,
  Row,
  Col,
  ListGroup,
  ListGroupItem,
  ToggleButton,
  ButtonToolbar,
  MenuItem,
  FormGroup,
  FormControl,
  PageHeader,
  Button,
  ControlLabel,
    Image
} from 'react-bootstrap';
import { CountyService, getUsersCounties } from '../../services';
import { UserService } from '../../services';


let countyService = new CountyService();
let userService = new UserService();

//henter infoen hvis personen er koblet til en kommune
export class Footer extends Component<{}> {


  state = {
      countyInformation:[],
     user: {}

  };




  componentDidMount = async ()=>{
    let user = {};
      await userService.getCurrentUser().then(r => {
        if (r[0] === undefined) {
          user = {};
        } else {
          user = r[0];
        }
        this.setState({user: user});
      });


    await countyService.getCountyEmployee(this.state.user.countyId).then((re: Array<Object>) => {
      this.setState({
        countyInformation: re
      });
    });
  };

  render() {
    let check;
    if (this.state.user.countyId !== {}) {
      return (
        <footer className="footerClass2">
          <Col xs={12} md={12}>
            <h4 className="footerInfo">Kontaktinformasjon til {this.state.user.county}-kommune:</h4>
            {
              this.state.countyInformation.map((r, i) => {
                return (

                  <div key={i} className="footerInfo">
                    <Row>kommuneansatt: {r.firstName + ' ' + r.lastName}</Row>
                    <Row> e-post: {r.mail}</Row>
                    <Row> telefonnr: {r.phone}</Row>
                  </div>
                );
              })
            }
          </Col>

          <div align="center">

            <Image className="picture"
                   src={'./resources/logo_white.png'}
                   rounded/>
            <h2>HVERDAGSHELT</h2>

          </div>

        </footer>
      );
    } else {
      return (
        <footer className="footerClass2">
          <div className="container">
            <div align="center">
              <Image className="picture"
                     src={'./resources/logo_white.png'}
                     rounded/>
              <h2>HVERDAGSHELT</h2>
            </div>
          </div>
        </footer>
      );
    }
    return (

      { check }

    );


  }

}
