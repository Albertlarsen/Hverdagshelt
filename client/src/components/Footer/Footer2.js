//@flow
import React, { Component } from 'react';
import {
  Grid,
  Row,
  Col,
  ListGroup,
  ListGroupItem,
  ToggleButton,
  ButtonToolbar,
  MenuItem,
  FormGroup,
  FormControl,
  PageHeader,
  Button,
  ControlLabel
} from 'react-bootstrap';
import { CountyService, getUsersCounties } from '../../services';
import { UserService } from '../../services';
import css from './Footer2.css';


let countyService = new CountyService();
let userService = new UserService();

//henter infoen hvis personen er koblet til en kommune
export class Footer2 extends Component<{}> {


  state = {
    countyInformation: [],
    user: []

  };


  componentDidMount = async () => {
    if (!(window.localStorage.getItem('userToken') == '')) {
      await userService.getCurrentUser().then(r => {
        this.setState({ user: r[0] });
      });
    }

    await countyService.getCountyEmployee(window.sessionStorage.getItem('countyId')).then((re: Array<Object>) => {
      this.setState({
        countyInformation: re[0]
      });
    });
  };


  render() {
    console.log('ils', (window.sessionStorage.getItem('countyName')));

    if (!(window.localStorage.getItem('userToken') == '') && !(window.sessionStorage.getItem('countyName') == 'undefined')) {

      let employee;
      employee = <div align="center">
        <h5>Kontaktinformasjon til {window.sessionStorage.getItem('countyName')} kommune:</h5>
        <ul>
          <li>
            <span>Kommuneansatt: {this.state.countyInformation.firstName + ' ' + this.state.countyInformation.lastName}</span>
          </li>
          <li><span>E-post: {this.state.countyInformation.mail}</span></li>
          <li><span>Telefonnr: {this.state.countyInformation.phone}</span></li>
        </ul>
      </div>;

      return (
        <footer id="myFooter">
          <Grid>
            <Row>

              <Col sm={4}>
                <div align="center">
                  <h5>Kom igang</h5>
                  <ul>
                    <li><a href="#">Hjem</a></li>
                    <li><a href="#">FAQ</a></li>
                    <li><a href="#">Kontakt oss</a></li>
                  </ul>
                </div>
              </Col>
              <Col sm={8}>
                {employee}
              </Col>

            </Row>

          </Grid>

          <div className="footer-copyright">
            <p>© 2019 Hverdagshelt </p>
          </div>
        </footer>
      );
    } else {
      return (
        <footer id="myFooter">
          <Grid>
            <Row>

              <div align="center">
                <h5>Kom igang</h5>
                <ul>
                  <li><a href="#">Hjem</a></li>
                  <li><a href="#">FAQ</a></li>
                  <li><a href="#">En ting til?</a></li>
                </ul>
              </div>

            </Row>

          </Grid>

          <div className="footer-copyright">
            <p>© 2019 Hverdagshelt </p>
          </div>
        </footer>
      );
    }
  }
}
