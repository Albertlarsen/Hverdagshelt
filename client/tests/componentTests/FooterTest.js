// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {Footer} from "../../src/components/Footer/Footer";

describe('Test for Footer component', () => {
    const wrapper = shallow(<Footer />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});