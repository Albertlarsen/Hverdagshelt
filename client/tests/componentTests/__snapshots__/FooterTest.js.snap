// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Footer component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Col
          bsClass="col"
          componentClass="div"
          md={12}
          xs={12}
        >
          <h4
            className="footerInfo"
          >
            Kontaktinformasjon til 
            -kommune:
          </h4>
        </Col>,
        <div
          align="center"
        >
          <Image
            bsClass="img"
            circle={false}
            className="picture"
            responsive={false}
            rounded={true}
            src="./resources/logo_svart.png"
            thumbnail={false}
          />
          <h2>
            HVERDAGSHELT
          </h2>
        </div>,
      ],
      "className": "footerClass2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <h4
              className="footerInfo"
            >
              Kontaktinformasjon til 
              -kommune:
            </h4>,
            Array [],
          ],
          "componentClass": "div",
          "md": 12,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Kontaktinformasjon til ",
                undefined,
                "-kommune:",
              ],
              "className": "footerInfo",
            },
            "ref": null,
            "rendered": Array [
              "Kontaktinformasjon til ",
              undefined,
              "-kommune:",
            ],
            "type": "h4",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "align": "center",
          "children": Array [
            <Image
              bsClass="img"
              circle={false}
              className="picture"
              responsive={false}
              rounded={true}
              src="./resources/logo_svart.png"
              thumbnail={false}
            />,
            <h2>
              HVERDAGSHELT
            </h2>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "img",
              "circle": false,
              "className": "picture",
              "responsive": false,
              "rounded": true,
              "src": "./resources/logo_svart.png",
              "thumbnail": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "HVERDAGSHELT",
            },
            "ref": null,
            "rendered": "HVERDAGSHELT",
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Col
            bsClass="col"
            componentClass="div"
            md={12}
            xs={12}
          >
            <h4
              className="footerInfo"
            >
              Kontaktinformasjon til 
              -kommune:
            </h4>
          </Col>,
          <div
            align="center"
          >
            <Image
              bsClass="img"
              circle={false}
              className="picture"
              responsive={false}
              rounded={true}
              src="./resources/logo_svart.png"
              thumbnail={false}
            />
            <h2>
              HVERDAGSHELT
            </h2>
          </div>,
        ],
        "className": "footerClass2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <h4
                className="footerInfo"
              >
                Kontaktinformasjon til 
                -kommune:
              </h4>,
              Array [],
            ],
            "componentClass": "div",
            "md": 12,
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Kontaktinformasjon til ",
                  undefined,
                  "-kommune:",
                ],
                "className": "footerInfo",
              },
              "ref": null,
              "rendered": Array [
                "Kontaktinformasjon til ",
                undefined,
                "-kommune:",
              ],
              "type": "h4",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "align": "center",
            "children": Array [
              <Image
                bsClass="img"
                circle={false}
                className="picture"
                responsive={false}
                rounded={true}
                src="./resources/logo_svart.png"
                thumbnail={false}
              />,
              <h2>
                HVERDAGSHELT
              </h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "img",
                "circle": false,
                "className": "picture",
                "responsive": false,
                "rounded": true,
                "src": "./resources/logo_svart.png",
                "thumbnail": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "HVERDAGSHELT",
              },
              "ref": null,
              "rendered": "HVERDAGSHELT",
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
